File: services/gui/app/models/title.js
const { Sequelize, DataTypes } = require("sequelize");
const sequelize = new Sequelize(process.env.DATABASE_URL);

const Title = sequelize.define("Title", {
  name: { type: DataTypes.STRING, allowNull: false },
  createdAt: { type: DataTypes.DATE, defaultValue: Sequelize.NOW },
  seen: { type: DataTypes.BOOLEAN, defaultValue: false },
  seenOn: { type: DataTypes.DATE },
  poster: { type: DataTypes.STRING },
  imdbRating: { type: DataTypes.FLOAT },
  imdbID: { type: DataTypes.STRING },
  year: { type: DataTypes.STRING },
  tomatoUserRating: { type: DataTypes.FLOAT },
  tomatoURL: { type: DataTypes.STRING },
  user: { type: DataTypes.STRING },
  genres: { type: DataTypes.ARRAY(DataTypes.STRING) }
});

module.exports = { Title };

File: services/gui/app/models/title.go
package models

import (
	"time"
)

// Title struct definition
type Title struct {
	Name            string    `json:"name" bson:"name" validate:"required"`
	CreatedAt       time.Time `json:"createdAt" bson:"createdAt" validate:"required"`
	Seen            bool      `json:"seen" bson:"seen"`
	SeenOn          time.Time `json:"seenOn" bson:"seenOn"`
	Poster          string    `json:"poster" bson:"poster"`
	ImdbRating      float64   `json:"imdbRating" bson:"imdbRating"`
	ImdbID          string    `json:"imdbID" bson:"imdbID"`
	Year            string    `json:"year" bson:"year"`
	TomatoUserRating float64  `json:"tomatoUserRating" bson:"tomatoUserRating"`
	TomatoURL       string    `json:"tomatoURL" bson:"tomatoURL"`
	User            string    `json:"user" bson:"user"`
	Genres          []string  `json:"genres" bson:"genres"`
}
